src/actions/notifications.rs

```diff
@@ -271,8 +271,24 @@ impl<'a> Action<'a> for DidChangeWatchedFiles {
 }
 
 impl<'a> NotificationAction<'a> for DidChangeWatchedFiles {
-    fn handle<O: Output>(&mut self, _params: DidChangeWatchedFilesParams, ctx: &mut ActionContext, out: O) -> Result<(), ()> {
+    fn handle<O: Output>(
+        &mut self,
+        params: DidChangeWatchedFilesParams,
+        ctx: &mut ActionContext,
+        out: O,
+    ) -> Result<(), ()> {
         trace!("on_cargo_change: thread: {:?}", thread::current().id());
+
+        // ignore irrelevant files from more spammy clients
+        if !params.changes.iter().any(|change| {
+                change.uri.as_str().ends_with("/Cargo.toml") ||
+                change.uri.as_str().ends_with("/Cargo.lock") ||
```

Comment:
I'm a bit worried about having these strings hard-coded here, and the corresponding patterns at https://github.com/rust-lang-nursery/rls/blob/master/src/actions/notifications.rs#L46, it seems like it would be an easy mistake to change one without the other. I would like them to come from a single source of truth. I think the easiest solution is to have an object or type with methods to provide the data.

Reply:
Yep that's probably a good shout, I'll take another look at it when I get time